app/models/manageiq/consumption/showback_rate.rb

```diff
@@ -24,51 +25,58 @@ class ShowbackRate < ApplicationRecord
 
     serialize :screener, JSON # Implement data column as a JSON
     default_value_for :screener, { }
+    validates :screener, :exclusion => { :in => [nil] }
 
     def name
       "#{category}:#{dimension}"
     end
 
-    def rate(value, event)
+    def rate(event)
       # Find tier (use context)
       # Calculate value within tier
       # For each tier used, calculate costs
-      rate_with_values(value || 0, event.time_span, event.month_duration)
+
+      # TODO event.resource.type should be eq to category
+      a = dimension.split('#')
+      value, measure = event.get_measure(a[0], a[1])
+      rate_with_values(value, measure,event.time_span, event.month_duration)
     end
 
-    def rate_with_values(value, time_span, cycle_duration, date = Time.now)
-      send(calculation.downcase, value || 0, time_span, cycle_duration, date)
+    def rate_with_values(value, measure, time_span, cycle_duration, date = Time.current)
+      send(calculation.downcase, value, measure, time_span, cycle_duration, date)
```

Comment:
maybe guard condition against `VALID_RATE_CALCULATIONS` ?

Reply:
I added the guard condition in the model. You can't have an instance that is not part of VALID_RATE_CALCULATIONS

Reply:
validates :calculation, :presence => true, :inclusion => { :in => VALID_RATE_CALCULATIONS }